### PR TITLE
[telemetry] allow a different remote destination for telemetry stats

### PR DESCRIPTION
**Pull request by truthbk**
This is useful for testing, will typically be unused by customers.

### CODE REVIEW

File: src/main/java/com/timgroup/statsd/NonBlockingStatsDClientBuilder.java [hunk 1]
```diff
@@ -129,17 +147,25 @@ public NonBlockingStatsDClientBuilder telemetryFlushInterval(int val) {
...
-                    addressLookup, timeout, socketBufferSize, maxPacketSizeBytes, entityID,
-                    bufferPoolSize, processorWorkers, senderWorkers, blocking, enableTelemetry,
-                    telemetryFlushInterval);
-        } else {
-            return new NonBlockingStatsDClient(prefix, queueSize, constantTags, errorHandler,
-                    staticStatsDAddressResolution(hostname, port), timeout, socketBufferSize, maxPacketSizeBytes,
-                    entityID, bufferPoolSize, processorWorkers, senderWorkers, blocking, enableTelemetry,
-                    telemetryFlushInterval);
+        Callable<SocketAddress> lookup = addressLookup;
+        Callable<SocketAddress> telemetryLookup = null;
```
Comment (ogaca-dd) [inline src/main/java/com/timgroup/statsd/NonBlockingStatsDClientBuilder.java:151]: `Callable<SocketAddress> telemetryLookup = telemetryAddressLookup`?

Comment (truthbk) [reply src/main/java/com/timgroup/statsd/NonBlockingStatsDClientBuilder.java:151]: Yes! Not sure what happened! 🤦

File: src/main/java/com/timgroup/statsd/NonBlockingStatsDClient.java [hunk 2]
```diff
@@ -281,10 +288,40 @@ public NonBlockingStatsDClient(final String prefix, final int queueSize, String[
...
-            statsDSender = createSender(addressLookup, handler, clientChannel,
-                    statsDProcessor.getBufferPool(), statsDProcessor.getOutboundQueue(), senderWorkers, this.telemetry);
+            this.telemetry = new Telemetry(telemetrytags, telemetryStatsDProcessor);
+
+            statsDSender = createSender(addressLookup, handler, clientChannel, statsDProcessor.getBufferPool(),
+                    statsDProcessor.getOutboundQueue(), senderWorkers, this.telemetry);
+            telemetryStatsDSender = statsDSender;
+
+            if (telemetryStatsDProcessor != statsDProcessor) {
+                telemetryStatsDSender = createSender(addressLookup, handler, clientChannel,
```
[the remark this answers is not in the document]
Comment (truthbk) [reply src/main/java/com/timgroup/statsd/NonBlockingStatsDClient.java]: Ah yes, I forgot the PR was actually incomplete 🤦 

I got it working on the deployment and then forgot fixing a few little things. Good catch. 

EDIT: I actually encountered something interesting when addressing this. If I create a new UDP channel with `telemetryClientChannel = DatagramChannel.open()` it won't work. I don't have a full explanation of why, but it just doesn't. So when `clientChannel` is UDP, we just reuse the same socket; and that's fine really. In any case it was something totally unexpected. I'll try to better understand why.

File: src/main/java/com/timgroup/statsd/NonBlockingStatsDClient.java [hunk 3]
```diff
@@ -281,10 +288,40 @@ public NonBlockingStatsDClient(final String prefix, final int queueSize, String[
...
+                        telemetryStatsDProcessor.getBufferPool(), telemetryStatsDProcessor.getOutboundQueue(),
+                        1, this.telemetry);
+            }
```
[the remark this answers is not in the document]
Comment (truthbk) [reply src/main/java/com/timgroup/statsd/NonBlockingStatsDClient.java:328]: Indeed!! I'm not even sure how that was working (but it was as we can see in the test CNAB deployments).

File: src/main/java/com/timgroup/statsd/NonBlockingStatsDClient.java [hunk 4]
```diff
@@ -281,10 +288,44 @@ public NonBlockingStatsDClient(final String prefix, final int queueSize, String[
...
+                // similar settings, but a single worker and non-blocking.
+                telemetryStatsDProcessor = createProcessor(queueSize, handler, maxPacketSizeBytes,
+                        poolSize, 1, false);
+            }
 
-            statsDSender = createSender(addressLookup, handler, clientChannel,
-                    statsDProcessor.getBufferPool(), statsDProcessor.getOutboundQueue(), senderWorkers, this.telemetry);
+            this.telemetry = new Telemetry(telemetrytags, telemetryStatsDProcessor);
+
+            statsDSender = createSender(addressLookup, handler, telemetryClientChannel, statsDProcessor.getBufferPool(),
```
Comment (ogaca-dd) [inline src/main/java/com/timgroup/statsd/NonBlockingStatsDClient.java:318]: Is it expected to use `telemetryClientChannel` and not `clientChannel`?

Comment (truthbk) [reply src/main/java/com/timgroup/statsd/NonBlockingStatsDClient.java]: I fixed the wrong line 🤦

**Pull request merged**